src/components/GlobalFilter/TagsModal.tsx

```diff
@@ -100,14 +107,15 @@ const TagsModal = ({
   const [sidsSelected, setSidsSelected] = useState<CommonSelectedTag[]>([]);
   const [filterBy, setFilterBy] = useState('');
   const [filterSIDsBy, setFilterSIDsBy] = useState('');
-  const dispatch = useDispatch();
+  const tagsData = useAtomValue(tagsAtom);
+  const sidsData = useAtomValue(sidsAtom);
   const [tagsLoaded, tagsCount, tagsPage, tagsPerPage] = useMetaSelector('tags');
   const [sidLoaded, sidCount, sidPage, sidPerPage] = useMetaSelector('sid');
-  const tags = useSelector<ReduxState, GlobalFilterTag[]>(({ globalFilter: { tags } }) => tags?.items || []);
-  const sids = useSelector<ReduxState, SID[]>(({ globalFilter: { sid } }) => sid?.items || []);
-  const filterScope = useSelector(({ globalFilter: { scope } }: ReduxState) => scope || undefined);
-  const debounceGetTags = useDebounce(fetchAllTags, tagsPerPage, selectedTags);
-  const debounceGetSIDs = useDebounce(fetchAllSIDs, sidPerPage, selectedTags);
+  const tags = tagsData.items;
+  const sids = sidsData.items;
```

Comment:
The new variables are not necessary, you can rename the `items` attribute while "deconstructing" the value of the atom:

```tsx

  const { items: tags } = useAtomValue(tagsAtom);
  const { items: sids } = useAtomValue(sidsAtom);
```